tweets/2022-04-27/2022-04-27-tweets.md

```diff
@@ -0,0 +1,342 @@
+🚨 Feature Alert 🚨
+
+Rocket improves the way it handles web components 🎉
+
+➡️ maintain a list of available tag names and it's imports
+➡️ automatically injects imports as needed
+➡️ enables progressive hydration 💪
+
+How does it work? 👇
+
+---
+
+First, we start by creating a very simple web components
+
+We call it `hello-wave` and it has some static HTML & CSS
+
+👉 `site/src/components/hello-wave.js`
+
+```js
+import { LitElement, html, css } from 'lit';
+
+export class HelloWave extends LitElement {
+  render() {
+    return html`<h2>Hello 👋</h2>`;
+  }
+
+  static styles = css`
+    h2 {
+      color: #e03131;
+    }
+  `;
+}
+```
+
+---
+
+We then add it to the rocket components object in the format of
+'tag-name': 'bare-import::ClassName'
+
+For this we are using a private import "#components/\*" which is mapped to the components folder
+
+Note: We need separate entrypoints for every class so we can load them individually
+
+👉 `data/pages/recursive.data.js`
+
+```js
+export const components = {
+  'hello-wave': '#components/HelloWave::HelloWave',
+};
+```
+
+👉 `package.json`
+
+```json
+{
+  "imports": {
+    "#components/*": "./site/src/components/*.js"
+  }
+}
+```
+
+---
+
+Now we can start using the component on our pages.
+
+➡️ right after we save
+➡️ it injects the component registration
+➡️ and the component gets server rendered
+➡️ to a Declarative Shadow DOM
+
+👉 `site/pages/index.rocket.js`
+
+```js
+export default () => html`
+  <h2>Hello World</h2>
+  <hello-wave></hello-wave>
+`;
+```
+
+```js
+/* START - Rocket auto generated - do not touch */
+export const sourceRelativeFilePath = 'index.rocket.js';
+import { html, components, layout } from './recursive.data.js';
+export { html, components, layout };
+export async function registerCustomElements() {
+  // server-only components
+  customElements.define('hello-wave', await import('#components/HelloWave').then(m => m.HelloWave));
```

Comment:
I was wondering the way this class is marked. Isn't the file itself called `hello-wave.js` ? Does this kind of a import work? (this is just me not being aware, if it does)